### ISSUE TITLE
Segfault on ToT darktable-3.2.x

### ISSUE THREAD

**Issue by hatsunearu** [darktable-org/darktable#7214]
**Describe the bug**
Segfault on top of tree darktable-3.2.x, release-3.2.1 on Linux Ubuntu 20.10. Probably related to corrupt config. 

**To Reproduce**
1. Launch darktable with bad config

Note: launching darktable with a different config via `--configdir` makes the problem not happen. 

**Platform (please complete the following information):**
 - Darktable Version: top of tree darktable-3.2.x, release-3.2.1
 - OS: Ubuntu 20.10, confirmed with Ubuntu version and self-compiled version
 - OpenCL was activated but I don't think that matters.

**Additional context**

[darktable_bt_BE34U0.txt](https://github.com/darktable-org/darktable/files/5648524/darktable_bt_BE34U0.txt)

I also ran valgrind on darktable:
[valgrind_output.txt](https://github.com/darktable-org/darktable/files/5648533/valgrind_output.txt)

The interesting bits are here:

``` 
==73282== Invalid write of size 8
==73282==    at 0x5057A892: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_impl_data::_Vector_impl_data() (stl_vector.h:95)
==73282==    by 0x50579059: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_impl::_Vector_impl() (stl_vector.h:130)
==73282==    by 0x505783E1: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_base() (stl_vector.h:285)
==73282==    by 0x50578401: std::vector<lfCamera*, std::allocator<lfCamera*> >::vector() (stl_vector.h:484)
==73282==    by 0x5057178B: lfDatabase::lfDatabase() (database.cpp:34)
==73282==    by 0x50487F1E: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282==  Address 0x1137db18 is 0 bytes after a block of size 40 alloc'd
==73282==    at 0x483CE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==73282==    by 0x50487F13: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282== 
==73282== Invalid write of size 8
==73282==    at 0x5057A89D: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_impl_data::_Vector_impl_data() (stl_vector.h:95)
==73282==    by 0x50579059: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_impl::_Vector_impl() (stl_vector.h:130)
==73282==    by 0x505783E1: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_base() (stl_vector.h:285)
==73282==    by 0x50578401: std::vector<lfCamera*, std::allocator<lfCamera*> >::vector() (stl_vector.h:484)
==73282==    by 0x5057178B: lfDatabase::lfDatabase() (database.cpp:34)
==73282==    by 0x50487F1E: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282==  Address 0x1137db20 is 8 bytes after a block of size 40 alloc'd
==73282==    at 0x483CE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==73282==    by 0x50487F13: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282== 
==73282== Invalid write of size 8
==73282==    at 0x5057A8A9: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_impl_data::_Vector_impl_data() (stl_vector.h:95)
==73282==    by 0x50579059: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_impl::_Vector_impl() (stl_vector.h:130)
==73282==    by 0x505783E1: std::_Vector_base<lfCamera*, std::allocator<lfCamera*> >::_Vector_base() (stl_vector.h:285)
==73282==    by 0x50578401: std::vector<lfCamera*, std::allocator<lfCamera*> >::vector() (stl_vector.h:484)
==73282==    by 0x5057178B: lfDatabase::lfDatabase() (database.cpp:34)
==73282==    by 0x50487F1E: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282==  Address 0x1137db28 is 16 bytes after a block of size 40 alloc'd
==73282==    at 0x483CE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==73282==    by 0x50487F13: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282== 
==73282== Invalid write of size 8
==73282==    at 0x5057A96E: std::_Vector_base<lfLens*, std::allocator<lfLens*> >::_Vector_impl_data::_Vector_impl_data() (stl_vector.h:95)
==73282==    by 0x5057913F: std::_Vector_base<lfLens*, std::allocator<lfLens*> >::_Vector_impl::_Vector_impl() (stl_vector.h:130)
==73282==    by 0x50578441: std::_Vector_base<lfLens*, std::allocator<lfLens*> >::_Vector_base() (stl_vector.h:285)
==73282==    by 0x50578461: std::vector<lfLens*, std::allocator<lfLens*> >::vector() (stl_vector.h:484)
==73282==    by 0x5057179B: lfDatabase::lfDatabase() (database.cpp:34)
==73282==    by 0x50487F1E: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282==  Address 0x1137db30 is 16 bytes after a block of size 48 in arena "client"
==73282== 
==73282== Invalid write of size 8
==73282==    at 0x5057A979: std::_Vector_base<lfLens*, std::allocator<lfLens*> >::_Vector_impl_data::_Vector_impl_data() (stl_vector.h:95)
==73282==    by 0x5057913F: std::_Vector_base<lfLens*, std::allocator<lfLens*> >::_Vector_impl::_Vector_impl() (stl_vector.h:130)
==73282==    by 0x50578441: std::_Vector_base<lfLens*, std::allocator<lfLens*> >::_Vector_base() (stl_vector.h:285)
==73282==    by 0x50578461: std::vector<lfLens*, std::allocator<lfLens*> >::vector() (stl_vector.h:484)
==73282==    by 0x5057179B: lfDatabase::lfDatabase() (database.cpp:34)
==73282==    by 0x50487F1E: init_global (lens.cc:1251)
==73282==    by 0x49B0C8D: dt_iop_load_module_so (imageop.c:436)
==73282==    by 0x49590E0: dt_module_load_modules (module.c:49)
==73282==    by 0x49B19A9: dt_iop_load_modules_so (imageop.c:1544)
==73282==    by 0x48F297E: dt_init (darktable.c:984)
==73282==    by 0x1090BB: main (main.c:92)
==73282==  Address 0x1137db38 is 24 bytes after a block of size 48 in arena "client"
```

I wanted to look at this myself since my config probably got messed up (dt crashed while importing some photos, and this started happening shortly after) but I gotta sleep for today... and I am not sure if my config has private info so I am withholding this until further guidance. Thanks.

**Comment by hatsunearu** [comment]
Haven't taken a very close look but it looks like I have a tag of length 0 in my database which is causing this issue. I have no idea how it got there (probably the crash, which should be another bug but I didn't capture any traces for that/lost the traces for that). For that crash I was importing a lot of photos from my NAS and it suddenly crashed.

**Comment by ptilopteri** [comment]
backup your library.db

run:   sqlite3 .config/darktable/library.db "PRAGMA INTEGRITY_CHECK"

if you believe you have a bad config, test it,
move ~/.config/darktable ~/.condig/XXdarktableXX

test darktable

to return to previous state:
exit darktable
rm -rf ~/.config/darktable
mv ~/.config/XXdarktableXX ~/.config/drktable

**Comment by johnny-bit** [comment]
After analyzing backtrace I managed to come up with a fix: #7599 - this however means that in all likelyhood db is corrupted or had weird writes.
